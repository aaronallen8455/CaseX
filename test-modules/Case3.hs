module Case3 where

main :: IO ()
main = pure ()

test :: Maybe Bool -> Bool -> Bool
test x y = case x of
  Nothing ->
    case y of
         False -> False
         True -> False
  Just _ -> True
