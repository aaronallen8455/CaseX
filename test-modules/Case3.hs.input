module Case3 where

main :: IO ()
main = pure ()

test :: Maybe Bool -> Bool -> Bool
test x y = case x of
  Nothing ->
    case y of
      SPLIT -> False
  Just _ -> True
