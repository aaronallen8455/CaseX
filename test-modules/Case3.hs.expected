module Case3 where

main :: IO ()
main = pure ()

test :: Maybe Bool -> Bool -> Bool
test x y = case x of
  -- 1
  Nothing ->
    case y of
      False -> False
      True -> False
  -- 2
  Just False -> True
  Just True -> True
  -- 3
